native/src/main/java/io/ballerina/stdlib/http/api/HttpUtil.java

```diff
@@ -1840,6 +1840,12 @@ private static void evaluateCommonFields(BMap<BString, Object> secureSocket, Ssl
         Parameter enableSessionCreationParam = new Parameter(HttpConstants.SECURESOCKET_CONFIG_SHARE_SESSION.getValue(),
                                                              enableSessionCreation);
         paramList.add(enableSessionCreationParam);
+        String sniHostName = "null".equals(String.valueOf(secureSocket.getStringValue(
```

Comment:
Rather than adding a `null` parameter can't we do something like this ?

```java
if (secureSocket.containsKey(HttpConstants.SECURESOCKET_CONFIG_SNI_HOST_NAME)) {
    // get the hostName and update parameter List
}
```

Reply:
Fixed [5fb2e7e](https://github.com/ballerina-platform/module-ballerina-http/pull/2235/commits/5fb2e7e6fdcde7e22e524b8f4a2674692ebd72cf)